### PR TITLE
add severity support for issues

### PR DESCRIPTION
**Pull request by alanrenmsft**
1. add severity support for validation issues
2. rename errors/messages to issues
3. use number input for scale and precision properties.
the changes in new STS version are all table designer related.

image

### CODE REVIEW

File: src/sql/workbench/browser/designer/designerIssuesTabPanelView.ts [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+		data.issueIcon = issueItem.appendChild(DOM.$(''));
+		data.issueText = issueItem.appendChild(DOM.$('.issue-text'));
+		return data;
+	}
+
+	renderElement(element: DesignerIssue, index: number, templateData: DesignerIssueListItemTemplate, height: number): void {
+		templateData.issueText.innerText = element.description;
+		templateData.issueText.title = element.description;
+		let iconClass;
+		switch (element.severity) {
```
Comment (alanrenmsft) [inline src/sql/workbench/browser/designer/designerIssuesTabPanelView.ts:105]: improvements to support different severity levels.

File: src/sql/azdata.proposed.d.ts [hunk 2]
```diff
@@ -1061,6 +1061,14 @@ declare module 'azdata' {
 		 */
 		export type DesignerEditPath = (string | number)[];
 
+		/**
+		 * Severity of the messages returned by the provider after processing an edit.
+		 * 'error': The issue must be fixed in order to commit the changes.
+		 * 'warning': Inform the user the potential risks with the current state. e.g. Having multiple edge constraints is only useful as a temporary state.
+		 * 'information': Informational message.
```
Comment (caohai) [inline src/sql/azdata.proposed.d.ts:1068]: 'informational' instead of 'information'?

**Review by caohai — APPROVED** [review]

File: src/sql/azdata.proposed.d.ts (hunk 2 above)
Comment (Charles-Gagnon) [reply src/sql/azdata.proposed.d.ts:1068]: I'd say 'information' for the string is more common anyways, but whatever one you prefer

File: src/sql/workbench/browser/designer/designerIssuesTabPanelView.ts [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+				iconClass = Codicon.info.classNames;
+				break;
+			default:
+				iconClass = Codicon.error.classNames;
+				break;
+		}
+		templateData.issueIcon.className = `issue-icon ${iconClass}`;
+	}
+
+	disposeElement?(element: DesignerIssue, index: number, templateData: DesignerIssueListItemTemplate, height: number): void {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/browser/designer/designerIssuesTabPanelView.ts:119]: Visibility modifier - and why is this optional here? That shouldn't be necessary in the implementation.

File: src/sql/workbench/services/tableDesigner/browser/tableDesignerComponentInput.ts [hunk 4]
```diff
@@ -372,15 +372,17 @@ export class TableDesignerComponentInput implements DesignerComponentInput {
 				description: localize('designer.column.description.precision', "For numeric data, the maximum number of decimal digits that can be stored in this database object."),
 				componentProperties: {
 					title: localize('tableDesigner.columnPrecisionTitle', "Precision"),
-					width: 60
+					width: 60,
+					inputType: 'number'
```
Comment (Charles-Gagnon) [inline src/sql/workbench/services/tableDesigner/browser/tableDesignerComponentInput.ts:376]: Were these intended to be in here?

**Review by Charles-Gagnon — APPROVED** [review]

File: src/sql/workbench/services/tableDesigner/browser/tableDesignerComponentInput.ts (hunk 4 above)
Comment (alanrenmsft) [reply src/sql/workbench/services/tableDesigner/browser/tableDesignerComponentInput.ts:376]: yes, I mentioned it in the PR description, just some small thing I noticed, so sliding in.

**Review by abist — APPROVED** [review]

File: src/sql/workbench/browser/designer/designerIssuesTabPanelView.ts (hunk 3 above)
Comment (alanrenmsft) [reply src/sql/workbench/browser/designer/designerIssuesTabPanelView.ts:119]: removed.

File: src/sql/workbench/browser/designer/designerIssuesTabPanelView.ts [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+	}
+
+	public disposeTemplate(templateData: DesignerIssueListItemTemplate): void {
```
Comment (abist) [inline src/sql/workbench/browser/designer/designerIssuesTabPanelView.ts:122]: is it ever disposed?

Comment (alanrenmsft) [reply src/sql/workbench/browser/designer/designerIssuesTabPanelView.ts:119]: the list component will call this.

**Pull request merged**